### DISCUSSION TITLE
Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project testmocks: Fatal error compiling

### DISCUSSION THREAD

**Discussion by YANGJINJUE** [apache/pulsar#20937] [Q&A] [answered]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

macos m1 12.6.6

### Minimal reproduce step

./mvnw package -Dmaven.test.skip

### What did you expect to see?

build success

### What did you see instead?

use jdk17, [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project testmocks: Fatal error compiling: 错误: 不支持发行版本 1.8 -> [Help 1]

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by nodece** [comment]
Do you use Java 1.8?

**Comment by YANGJINJUE** [comment]
> Do you use Java 1.8?

java version "17.0.2" 2022-01-18 LTS
Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)

**Comment by tisonkun** [comment]
I can build from source on master. This is likely an environment issue.

> 不支持发行版本 1.8

So when you execute the command, it resolves to JDK 1.8. Please check your environment or provide complete build log.

**Reply by YANGJINJUE** [reply, marked as answer]
The two environment variables, maven.compiler.target and maven.compiler.source, will be overridden by my local ~/.m2/settings.xml file. This issue has been resolved, and I am not using my local settings.xml anymore.
